lib/net/ssh/transport/algorithms.rb

```diff
@@ -44,11 +44,12 @@ class Algorithms
                   diffie-hellman-group14-sha256
                   diffie-hellman-group14-sha1],
 
-          encryption: %w[aes256-ctr aes192-ctr aes128-ctr],
+          encryption: %w[aes256-gcm@openssh.com aes128-gcm@openssh.com aes256-ctr aes192-ctr aes128-ctr],
```

Comment:
I've not had any good amount of time to look into this but it's this line that is causing the issue; changing to anything else (or removing) allows the test helper to start a new SSH connection. Unsure what the issue is, will try to assist further when I can.

Note: we're _very_ interested in getting support for aes-xxx-gcm ASAP.

Reply:
Note, changing the preference order allows the CI timeouts to pass:

```bash
integration % git diff
diff --git a/lib/net/ssh/transport/algorithms.rb b/lib/net/ssh/transport/algorithms.rb
index 6fc339b9..48472413 100644
--- a/lib/net/ssh/transport/algorithms.rb
+++ b/lib/net/ssh/transport/algorithms.rb
@@ -44,7 +44,8 @@ module Net
                   diffie-hellman-group14-sha256
                   diffie-hellman-group14-sha1],

-          encryption: %w[aes256-gcm@openssh.com aes128-gcm@openssh.com aes256-ctr aes192-ctr aes128-ctr],
+          encryption: %w[aes256-ctr aes192-ctr aes128-ctr
+                         aes256-gcm@openssh.com aes128-gcm@openssh.com],
```

Reply:
Applied it, waiting a maintener to run workflows. @fwininger @mfazekas

Reply:
You've removed multiple methods from `AEADAESGCM` which are causing the build to hard fail now. Was that intentional? https://github.com/net-ssh/net-ssh/compare/6d1f07286332486d133fbe22d4f43c1164c073f5..bb6f8fc3602dbe6754a436b616b6aaeee100767f

Reply:
Just applied rubocop. Gonna look it tomorrow.